### ISSUE TITLE
board.GetLegalMoves(true) makes IsStalemate behavior wrong

### ISSUE THREAD

**Issue by EikeSchwass** [SebLague/Chess-Challenge#418]
If you call the ````board.GetLegalMoves(true)```` method and it returns 0 moves (although non captures would be available), ````board.HasZeroLegalMoves()```` uses the cached ````GetLegalMoves(true)```` response and returns true.

**Comment by SebLague** [comment]
Should be fixed now, thank you! (I assume you were using the NonAlloc version of GetLegalMoves because I couldn't find an error with the regular version?)

**Comment by EikeSchwass** [comment]
> Should be fixed now, thank you! (I assume you were using the NonAlloc version of GetLegalMoves because I couldn't find an error with the regular version?)

Yes, sorry! I assumed the error would be the same in both GetLegalMoves and GetLegalMovesNonAlloc, should have been more rigorous in my bug report. Thanks for fixing it :)
